watchers/metrics.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package watchers
+
+import (
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/o11y/metrics"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// WatcherMetricsHandler is in charge of watcher metrics.
+type WatcherMetricsHandler interface {
```

Comment:
I'm not sure about calling this a handler; it's confusing being next to the actual (and required) kubernetes Handler terminology and APIs.

I like the idea of factorising even though it adds boilerplate -- however it feels more like a wrapper than a handler. Maybe it should be called  something like `WatcherMetricsWrapper` ?

Reply:
Completely agree with you 👍 
It is an adapter, so what do you think of `WatcherMetricsAdapter`?